### PR TITLE
SQL Server PKINIT configuration script

### LINKED ISSUE #18555: Add webAPI endpoints for SQLServer access AD configuration
## What: 
Similar to desktop access https://github.com/gravitational/teleport/blob/ba9fd70a92dcf1931aae26206351a885163048e7/lib/web/apiserver.go#L475 SQLServer AD setup should be automated for Database Access and Teleport should provide ps1 scripts that will allow to automate Windows Active Directory Setup.

### PR DESCRIPTION
**Pull request by gabrielcorado**
Closes #18555.

Adds a `webapi` endpoint that serves a PowerShell script used to configure SQL Server PKINIT on a Windows instance. The script is similar to the [Desktop Access script](https://github.com/gravitational/teleport/blob/master/lib/web/scripts/desktop/configure-ad.ps1):
1. Creates a GPO with SmartCard enabled and publishes the CA/CRL.
2. Generates a Database Service configuration contents.

<details>
<summary>How it was tested</summary>

Required setup:
* Windows machine joined an AD domain. The logged user needs enough permissions to add/edit GPOs.
* SQL Server instance joined the same AD domain.

Generate the script:
1. Create a provision token (join token): `tctl tokens add --type=db --ttl 1h`
2. Download the script contents from:
   ```
   /webapi/scripts/database-access/configure/sqlserver/<token>/configure-ad.ps1?uri=<uri>
   ```
   Replace the `<token>` with your provision token and `<uri>` with the database URI (including the port). For example:
   ```
   https://teleport.dev/webapi/scripts/database-access/configure/sqlserver/3d586dcd9797b004ae492838a2907059/configure-ad.ps1?uri=EC2AMAZ-XXXXXXX.ad.teleport.dev:1433
   ```
3. Execute the script on your Windows machine.
4. Copy the YAML output and run `teleport start` on a Linux machine. Note: the `kinit` with PKINIT extensions should be installed (`sudo apt-get -y install krb5-user krb5-pkinit`)

OBS: If your Windows machine has access, you can download and execute the script using the following:
```powershell
(Invoke-WebRequest -Uri ttps://teleport.dev/webapi/scripts/database-access/configure/sqlserver/3d586dcd9797b004ae492838a2907059/configure-ad.ps1?uri=EC2AMAZ-XXXXXXX.ad.teleport.dev:1433).Content | Invoke-Expression
```

</details>

### CODE REVIEW

File: lib/web/databases.go [hunk 1]
```diff
@@ -280,6 +284,76 @@ func (h *Handler) handleDatabaseGetIAMPolicy(w http.ResponseWriter, r *http.Requ
...
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	caCRL, err := h.GetProxyClient().GenerateCertAuthorityCRL(r.Context(), types.DatabaseCA)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if len(certAuthority.GetActiveKeys().TLS) != 1 {
```
[the remark this answers is not in the document]
Comment (fspmarshall) [reply lib/web/databases.go:332]: +1 we generally trust sets of cas rather than individual CAs.  I _think_ this is a requirement of the hsm feature.

**Review by fspmarshall — APPROVED** [review]
LGTM once exiting feedback is addressed, tho note that I'm not qualified to give feedback on powershell. Should make sure at least one other ps fluent person reviews that part.

File: lib/web/databases.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ibeckermayer) [reply lib/web/databases.go:332]: Actually I was mistaken, I do find similar checks in a couple of places

https://github.com/gravitational/teleport/blob/a4d88191e3478c47ae5c8e5d3de1a7d3476f704a/lib/web/desktop.go#L511-L513

https://github.com/gravitational/teleport/blob/a4d88191e3478c47ae5c8e5d3de1a7d3476f704a/lib/client/ca_export.go#L256-L258

I don't have an understanding of why we'd trust a set vs demand a single CA in different contexts. Do you know @gabrielcorado?

File: lib/web/apiserver.go [hunk 2]
```diff
@@ -635,6 +635,7 @@ func (h *Handler) bindDefaultEndpoints() {
 	h.PUT("/webapi/sites/:site/databases/:database", h.WithClusterAuth(h.handleDatabaseUpdate))
 	h.GET("/webapi/sites/:site/databases/:database", h.WithClusterAuth(h.clusterDatabaseGet))
 	h.GET("/webapi/sites/:site/databases/:database/iam/policy", h.WithClusterAuth(h.handleDatabaseGetIAMPolicy))
+	h.GET("/webapi/scripts/database-access/configure/sqlserver/:token/configure-ad.ps1", httplib.MakeHandler(h.sqlServerConfigureADScriptHandle))
```
Comment (zmb3) [inline lib/web/apiserver.go:638]: FWIW, I wish we hadn't used `/desktop-access/` in the URL for the desktop scripts. We change product names/branding all the time, I would have preferred something more product-agnostic.

While it's too late for desktops, you can still avoid this and use something like `/webapi/scripts/databases/configure/...`

**Review by zmb3 — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

**Pull request merged**